packages/sanity/playwright-ct/tests/formBuilder/inputs/PortableText/FocusTracking.spec.tsx

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable max-nested-callbacks */
+import {expect, test} from '@playwright/experimental-ct-react'
+import React from 'react'
+import {Path, SanityDocument} from '@sanity/types'
+import {Page} from '@playwright/test'
+import FocusTrackingStory from './FocusTrackingStory'
+
+export type UpdateFn = () => {focusPath: Path; document: SanityDocument}
+
+const document: SanityDocument = {
+  _id: '123',
+  _type: 'test',
+  _createdAt: new Date().toISOString(),
+  _updatedAt: new Date().toISOString(),
+  _rev: '123',
+  body: [
+    {
+      _type: 'block',
+      _key: 'a',
+      children: [{_type: 'span', _key: 'b', text: 'Foo'}],
+      markDefs: [],
+    },
+    {
+      _type: 'block',
+      _key: 'c',
+      children: [{_type: 'span', _key: 'd', text: 'Bar'}],
+      markDefs: [],
+    },
+    {
+      _type: 'block',
+      _key: 'e',
+      children: [{_type: 'span', _key: 'f', text: 'Baz'}],
+      markDefs: [],
+    },
+    {
+      _type: 'block',
+      _key: 'g',
+      children: [
+        {_type: 'span', _key: 'h', text: 'Hello '},
+        {_type: 'inlineObjectWithTextProperty', _key: 'i', text: 'there'},
+        {_type: 'span', _key: 'j', text: ' playwright'},
+      ],
+      markDefs: [],
+    },
+    {
+      _type: 'testObjectBlock',
+      _key: 'k',
+      text: 'Hello world',
+    },
+  ],
+}
+
+test.describe('Portable Text Input', () => {
+  test.beforeEach(async ({page}) => {
+    await page.evaluate(() => {
+      window.localStorage.debug = 'sanity-pte:*'
+    })
+  })
+  test.describe('Should track focusPath', () => {
+    test(`for span .text`, async ({mount, page}) => {
+      const initialPath = ['body', {_key: 'c'}, 'children', {_key: 'd'}, 'text']
+      const component = await mount(
+        <FocusTrackingStory document={document} focusPath={initialPath} />,
+      )
+      await waitForFocusPath(page, initialPath)
```

Comment:
**question:** is it a smell that `waitForFocusPath` checks for a data attribute on the corresponding _story wrapper_ rather than the underlying PTE implementation?

More specifically, shouldn't `window.getSelection()` return the correct value immediately after mounting if `focusPath` is provided?

Reply:
The editor will sync the value and eventually put the focusPath in effect in the editor through hooks in an asynchronous way. That's what makes this a bit tricky to test, and why I wrote the "helper" code to confirm that the update to the focusPath has been propagated to the browser before we continue the test.